src/store/modules/conversations.js

```diff
@@ -1,243 +1,119 @@
+    async setActive ({ commit, dispatch }, conversation) {
+      commit('setActive', { conversationId: conversation.id })
+      commit('setConversation', { conversation })
+      await dispatch('fetch', conversation.id)
+    },
 
-    let data
-    try {
-      data = await messageAPI.listMore(currentCursor)
-    }
-    catch (error) {
-      commit(types.RECEIVE_MORE_MESSAGES_ERROR, { conversationId, error })
-      return
-    }
-    commit(types.RECEIVE_MORE_MESSAGES, { conversationId, messages: data.results, cursor: data.next })
-  },
-}
+    async clearActive ({ commit }) {
+      commit('clearActive')
+    },
 
-export const mutations = {
-  [types.SET_ACTIVE] (state, { conversationId }) {
-    state.activeConversationId = conversationId
-  },
-  [types.CLEAR_ACTIVE] (state) {
-    state.activeConversationId = null
-  },
+    async receiveMessage ({ commit, state, getters }, message) {
+      // only add if messages doesn't exist yet
 
-  // receive initial messages
-  [types.REQUEST_MESSAGES] (state, { conversationId }) {
-    state.receiveStatus = {
-      isWaiting: true,
-      error: null,
-      success: false,
-    }
-  },
-  [types.RECEIVE_MESSAGES] (state, { conversationId, messages, cursor }) {
-    if (state.messages[conversationId]) {
-      Vue.set(state.messages, conversationId, messages)
-      Vue.set(state.cursors, conversationId, cursor)
-    }
-    if (state.activeConversationId === conversationId) {
-      state.receiveStatus = {
-        isWaiting: false,
-        error: null,
-        success: true,
-      }
-    }
-  },
-  [types.RECEIVE_MESSAGES_ERROR] (state, { conversationId, error }) {
-    console.log(error)
-    if (state.activeConversationId === conversationId) {
-      state.receiveStatus = {
-        isWaiting: false,
-        error,
-        success: false,
+      if (!getters.activeMessages.find(e => e.id === message.id)) {
+        commit('prependMessage', { message })
       }
-    }
-  },
-
-  // receive more messages
-  [types.REQUEST_MORE_MESSAGES] (state, { conversationId }) {
-    state.receiveMoreStatus = {
-      isWaiting: true,
-      error: null,
-    }
+    },
   },
-  [types.RECEIVE_MORE_MESSAGES] (state, { conversationId, messages, cursor }) {
-    // append at end
-    if (state.messages[conversationId]) {
-      state.messages[conversationId].push(...messages)
+  mutations: {
+    setActive (state, { conversationId }) {
+      state.activeConversationId = conversationId
+    },
+    clearActive (state) {
+      state.activeConversationId = null
+    },
+    setMessages (state, { conversationId, messages, cursor }) {
+      Vue.set(state.messages, conversationId, messages)
       Vue.set(state.cursors, conversationId, cursor)
-    }
-    if (state.activeConversationId === conversationId) {
-      state.receiveMoreStatus = {
-        isWaiting: false,
-        error: null,
+    },
+    appendMessages (state, { conversationId, messages, cursor }) {
```

Comment:
Not sure I quite understand why we would prepend and append messages.

Reply:
prepend: when the user writes a message and it should go to top
append: when loading more message history from the server

will add comments :)